ddtrace/contrib/influx/patch.py

```diff
@@ -0,0 +1,83 @@
+import influxdb  # version >=5.0
+import wrapt
+from influxdb.exceptions import InfluxDBClientError, InfluxDBServerError
+
+from . import metadata
+
+from ...utils.wrappers import unwrap
+from ...compat import urlencode
+from ...pin import Pin
+from ...ext import http, db, AppTypes
+
+
+DEFAULT_SERVICE = 'influxdb'
+SPAN_TYPE = 'sql'
+
+
+def patch():
+    if getattr(influxdb, '_datadog_patch', False):
+        return
+    setattr(influxdb, '_datadog_patch', True)
+    wrapt.wrap_function_wrapper('influxdb.client', 'InfluxDBClient.request', _request)
+
+    Pin(service=DEFAULT_SERVICE, app=DEFAULT_SERVICE, app_type=AppTypes.db).onto(influxdb.client.InfluxDBClient)
+
+
+def unpatch():
+    if getattr(influxdb, '_datadog_patch', False):
+        setattr(influxdb, '_datadog_patch', False)
+        unwrap(influxdb.client.InfluxDBClient, 'request')
+
+
+def _request(func, instance, args, kwargs):
+    """
+    Trace a request to InfluxDB at the level of a (retried) HTTP request.
+    :param func: A reference to influxdb.client.InfluxDBClient.request, called.
+    :param instance: An instance of influxdb.client.InfluxDBClient.
+    :param args: Arguemnts passed to influxdb.client.InfluxDBClient.request()
+    :param kwargs: Keyword arguments passed to influxdb.client.InfluxDBClient.request.
+    :return: An instance of requests.models.Response
+    """
+
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    with pin.tracer.trace('influx.request') as span:
+        # Don't instrument if the trace is not sampled
+        if not span.sampled:
+            return func(*args, **kwargs)
+
+        span.service = pin.service
+        span.span_type = SPAN_TYPE
+
+        url = kwargs.get('url')
+        span.set_tag(http.URL, url)
+
+        params = kwargs.get('params', {})
+
+        database = params.get('db', None)
+        if database:
+            span.set_tag(db.NAME, database)
+
+        method = kwargs.get('method', 'GET').upper()
+        if method == 'GET' and url.startswith('query'):
+            span.resource = params.get('q', None)
+        elif url:
+            span.resource = url
+
+        span.set_tag(http.METHOD, method)
+        span.set_tag(metadata.PARAMS, urlencode(params))
+
+        try:
+            result = func(*args, **kwargs)
+        except InfluxDBClientError as e:
```

Comment:
We have an API for errors that can be used to enhance the trace with a stack-trace and exception information. 

See the following for an example:
https://github.com/DataDog/dd-trace-py/blob/5a65259023bd7f846e913c5e23a94d1f5ad45f74/ddtrace/contrib/pymemcache/client.py#L157-L166

Reply:
Actually, you can ignore my comment here. Our Span API handles this already (when used as a context manager):
https://github.com/DataDog/dd-trace-py/blob/b727a05908f2fe9095b5dc0cf22558095d798ece/ddtrace/span.py#L298-L304